src/modules/api/pipeline-task/pipelines/utils.py

```diff
@@ -36,131 +26,89 @@ def load_json_from_request(request):
+          We'd have to explicitly handle the multiple value case, and only return 200 OK when all are complete.
   '''
-  task_id = payload.get('id', 'no-id')
-  logger.info(f"[TASK {task_id}] Starting job in queue { task_route }. Payload: { payload }")
-
-
-  # With exponential backoff, this is approx (2^(n-1))-1 = 127 sec, or a little over 2 minutes
-  max_tries = 8
 
-  def log_backoff(details):
-    logger.warn(f'[TASK {task_id}] Failed to run job on attempt {details["tries"]}/{max_tries}. Trying again in {details["wait"]:00.1f}s...')
+  # Try parsing from operation name for each subclass, skipping if name is invalid for that subclass
+  results = []
+  for cls in pipeline_subclasses:
+    try:
+      results.append( cls._Runner_Class.from_operation_name(operation_name) )
+    except:
+      pass
 
-  def log_giveup(details):
-    logger.warn(f'[TASK {task_id}] Failed to run job on attempt {details["tries"]}/{max_tries}. Total time elapsed: {details["elapsed"]:00.1f}s.')
+  # If none of the subclasses accepted the operation name, it is invalid
+  if len(results) == 0:
+    raise APIBadRequestError(f'Operation "{operation_name}" NOT FOUND -- nothing to do')
 
-  # Local helper function to ensure job is started
-  # There is a delay between when the job "create" request is sent and when the job becomes available for "run",
-  # so we use exponential backoff to wait for the job to finish being created
-  @backoff.on_exception(
-      backoff.expo, HttpError, giveup=lambda ex: ex.status_code != 400, max_tries=max_tries, on_backoff=log_backoff, on_giveup=log_giveup, jitter=None
-  )
-  def _run_job(handler):
-    return handler.run_job()
+  # If more than one subclass accepted, there was a name collision(!)
+  elif len(results) > 1:
+    raise APIBadRequestError(f'Operation "{operation_name}" matched multiple jobs')
 
-
-  # Try to create a task handler of the appropriate type
-  handler = get_task_handler(task_route, **payload)
-
-  # Create a CloudRun job for this task
+  # Try getting an execution ID from the operation name, defaulting to None
   try:
-    create_response = handler.create_job()
-
-  # If the job already exists, optionally bail out
-  except HttpError as ex:
-    if run_if_exists and ex.status_code == 409:
-      logger.warn(f'[TASK {task_id}] Encountered HttpError: {ex}')
-      logger.warn(f'[TASK {task_id}] Running job again...')
-      create_response = ex.resp
-    else:
-      raise
+    exec_id = results[0].get_execution_id_from_operation_name( operation_name )
+  except ValueError:
+    exec_id = None
 
-  # Run the CloudRun job
-  run_response, pub_sub_id = _run_job(handler)
+  # Return the one matching subclass object, and the execution ID
+  return results[0], exec_id
 
-  # Return all computed values
-  return handler, create_response, run_response
 
 
-
-def update_status_safe(queue_name, op_id, call_id='', status=None, **kwargs):
+def update_status_safe(job: JobPipeline, status: JobStatus, call_id: str):
```

Comment:
By using the `JobPipeline` object directly, we can avoid defining a bunch of job-specific update functions in the `service` subdirectory.